### ISSUE TITLE
Generic MultipleStatementAlignment sniff reports incorrect errors for multiple assignments on a single line

### ISSUE THREAD

**Issue by meijerm15** [squizlabs/PHP_CodeSniffer#564]
PHPCBF is currently aborting all changes to a file when encountering an unfix-able violation, like demonstrated below. Which is an A = B = C assignment.

This will cause it to not fix any violations anywhere else in this specific file, which makes it rather impossible to find this aborting violation in a huge class.

Could we get a line number/range of where this error/aborting violation is found or maybe if possible, this violation being fixed?

**Example**
```php
<?php

namespace test\test;

class Test
{
    public function foo()
    {
        $test = "test";
        $foo = $moo = $test;
    }

    public function moo()
    {
        $test = "test";
        $foo = $test;
    }
}
```
**PHPCS log**
```php
  9 | ERROR | [x] Equals sign not aligned with surrounding assignments;
    |       |     expected 7 spaces but found 1 space
    |       |     (Generic.Formatting.MultipleStatementAlignment.NotSame)
 10 | ERROR | [x] Equals sign not aligned with surrounding assignments;
    |       |     expected 8 spaces but found 1 space
    |       |     (Generic.Formatting.MultipleStatementAlignment.NotSame)
 16 | ERROR | [x] Equals sign not aligned with surrounding assignments;
    |       |     expected 2 spaces but found 1 space
    |       |     (Generic.Formatting.MultipleStatementAlignment.NotSame)
```
**PHPCBF log**
```php
Processing Test.php [PHP => 85 tokens in 18 lines]... DONE in 4ms (3 fixable violations)
        => Fixing file: 2/3 violations remaining [made 50 passes]... ERROR in 505ms
No fixable errors were found
Time: 533ms; Memory: 5.75Mb
```
**Version**
```php
PHP_CodeSniffer version 2.3.0 (stable) by Squiz (http://www.squiz.net)
```

**Comment by aik099** [comment]
Please also specify command you're using to execute `phpcbf` with all arguments.

**Comment by meijerm15** [comment]
I have used a custom standard with a custom ruleset, the related rules to the report is written below.

**Command**
```php
phpcbf -n --standard=x
```
**Ruleset**
```xml
<rule ref="Generic.Formatting.DisallowMultipleStatements"/>
<rule ref="Generic.Formatting.MultipleStatementAlignment"/>
<rule ref="Generic.Formatting.MultipleStatementAlignment">
    <properties><property name="ignoreMultiLine" value="true"/>
    <property name="error" value="true"/></properties>
</rule>
```

**Comment by meijerm15** [comment]
*Update:*

Default standard: **MySource** and **PHPCS** have 2 states of trying to fix this violation.

**1**
```php
<?php

namespace test\test;

class Test
{


    public function foo()
    {
        $test = 'test';
        $foo         = $moo = $test;

    }//end foo()


    public function moo()
    {
        $test = 'test';
        $foo  = $test;

    }//end moo()


}//end class
```
**2**
```php
<?php

namespace test\test;

class Test
{


    public function foo()
    {
        $test               = 'test';
        $foo   = $moo = $test;

    }//end foo()


    public function moo()
    {
        $test = 'test';
        $foo  = $test;

    }//end moo()


}//end class
```

**Comment by gsherwood** [comment]
The easiest way to debug a PHPCBF error is using the PHPCS diff report with either -vv or -vvv on the command line.

As an example, take this very simple file:
```php
<?php
$foo = $moo = $test;
```
Then run `phpcs temp.php --standard=Squiz --report=diff -vv` and you get this towards the end:
```
* fixed 1 violations, starting loop 48 *
E: [Line 2] Equals sign not aligned correctly; expected 1 space but found 15 spaces (Generic.Formatting.MultipleStatementAlignment.Incorrect)
Generic_Sniffs_Formatting_MultipleStatementAlignmentSniff (line 302) replaced token 2 (T_WHITESPACE) "···············=" => "·="
* fixed 1 violations, starting loop 49 *
E: [Line 2] Equals sign not aligned with surrounding assignments; expected 8 spaces but found 1 space (Generic.Formatting.MultipleStatementAlignment.NotSame)
Generic_Sniffs_Formatting_MultipleStatementAlignmentSniff (line 302) replaced token 2 (T_WHITESPACE) "·=" => "········="
* fixed 1 violations, starting loop 50 *
E: [Line 2] Equals sign not aligned with surrounding assignments; expected 15 spaces but found 8 spaces (Generic.Formatting.MultipleStatementAlignment.NotSame)
Generic_Sniffs_Formatting_MultipleStatementAlignmentSniff (line 302) replaced token 2 (T_WHITESPACE) "········=" => "···············="
* fixed 1 violations, starting loop 51 *
*** Reached maximum number of loops with 1 violations left unfixed ***
*** END FILE FIXING ***
```
This shows that the `Generic.Formatting.MultipleStatementAlignment` sniff is fighting with itself, so the fixing in there must be wrong.

If you use -vvv, you also get the file contents after each run so you can more clearly see what is going on:
```
	* fixed 1 violations, starting loop 48 *
---START FILE CONTENT---
1|<?php
2|$foo               = $moo = $test;
3|
--- END FILE CONTENT ---
	E: [Line 2] Equals sign not aligned correctly; expected 1 space but found 15 spaces (Generic.Formatting.MultipleStatementAlignment.Incorrect)
	Generic_Sniffs_Formatting_MultipleStatementAlignmentSniff (line 302) replaced token 2 (T_WHITESPACE) "···············=" => "·="
	* fixed 1 violations, starting loop 49 *
---START FILE CONTENT---
1|<?php
2|$foo = $moo = $test;
3|
--- END FILE CONTENT ---
	E: [Line 2] Equals sign not aligned with surrounding assignments; expected 8 spaces but found 1 space (Generic.Formatting.MultipleStatementAlignment.NotSame)
	Generic_Sniffs_Formatting_MultipleStatementAlignmentSniff (line 302) replaced token 2 (T_WHITESPACE) "·=" => "········="
	* fixed 1 violations, starting loop 50 *
---START FILE CONTENT---
1|<?php
2|$foo        = $moo = $test;
3|
--- END FILE CONTENT ---
	E: [Line 2] Equals sign not aligned with surrounding assignments; expected 15 spaces but found 8 spaces (Generic.Formatting.MultipleStatementAlignment.NotSame)
	Generic_Sniffs_Formatting_MultipleStatementAlignmentSniff (line 302) replaced token 2 (T_WHITESPACE) "········=" => "···············="
	* fixed 1 violations, starting loop 51 *
	*** Reached maximum number of loops with 1 violations left unfixed ***
	*** END FILE FIXING ***
```
I'll take a look at fixing the sniff now.

**Comment by gsherwood** [comment]
I've found and fixed the problem. The sniff needed to only check the first assignment on a line, but it wasn't. So it was also throwing incorrect errors as well as causing failures during fixing.
